core/src/main/java/com/linecorp/armeria/internal/TransportType.java

```diff
@@ -0,0 +1,105 @@
+package com.linecorp.armeria.internal;
+
+import static com.linecorp.armeria.common.util.NativeLibraries.isEpollAvailable;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.epoll.EpollDatagramChannel;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.epoll.EpollSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.DatagramChannel;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+
+/**
+ * Native transport types.
+ */
+public enum TransportType {
+    NIO(NioServerSocketChannel.class, NioSocketChannel.class, NioDatagramChannel.class,
+        NioEventLoopGroup.class, NioEventLoopGroup::new),
+
+    EPOLL(EpollServerSocketChannel.class, EpollSocketChannel.class, EpollDatagramChannel.class,
+          EpollEventLoopGroup.class, EpollEventLoopGroup::new);
+
+    private final Class<? extends ServerChannel> serverChannelClass;
+    private final Class<? extends SocketChannel> socketChannelClass;
+    private final Class<? extends DatagramChannel> datagramClass;
+    private final Class<? extends EventLoopGroup> eventLoopGroupClass;
+    private final BiFunction<Integer, ThreadFactory, ? extends EventLoopGroup> eventLoopGroupConstructor;
+
+    TransportType(Class<? extends ServerChannel> serverChannelClass,
+                  Class<? extends SocketChannel> socketChannelClass,
+                  Class<? extends DatagramChannel> datagramClass,
+                  Class<? extends EventLoopGroup> eventLoopGroupClass,
+                  BiFunction<Integer, ThreadFactory, ? extends EventLoopGroup> eventLoopGroupConstructor) {
+        this.serverChannelClass = serverChannelClass;
+        this.socketChannelClass = socketChannelClass;
+        this.datagramClass = datagramClass;
+        this.eventLoopGroupClass = eventLoopGroupClass;
+        this.eventLoopGroupConstructor = eventLoopGroupConstructor;
+    }
+
+    /**
+     * Returns the {@link ServerChannel} class that is avaiable for this transport type.
+     */
+    public Class<? extends ServerChannel> serverChannelClass() {
+        return serverChannelClass;
+    }
+
+    /**
+     * Returns the available {@link EventLoopGroup}.
+     */
+    public EventLoopGroup eventLoopGroup(int nThreads,
```

Comment:
Just naming concern, because this method will `create` eventLoopGroup when calling, add `create` may better ?